assets/src/edit-story/components/popup/utils.js

```diff
@@ -144,10 +144,10 @@ export function getOffset(placement, spacing, anchor, dock, popup) {
     dockRect,
     bodyRect
   );
-  const maxOffsetX = bodyRect.width - spacingH;
+  const maxOffsetX = bodyRect.width - width - getXTransforms(placement) * width;
```

Comment:
I wasn't sure what the `spacingH` was for, but this properly set the element to a max of the viewport edge. Happy to add `spacingH` back in if needed